pkg/virtualization/core/virtualsystem/virtualmachine.go

```diff
@@ -943,6 +1104,43 @@ func (vm *VirtualMachine) GetVirtualHardDiskByPath(path string) (vhd *disk.Virtu
 	return
 }
 
+func (vm *VirtualMachine) GetAttachedVirtualHardDisks() (vhdPaths []string, err error) {
+	col, err := vm.GetVirtualHardDisks()
+	if err != nil {
+		return
+	}
+	defer col.Close()
+
+	for _, inst := range col {
+		tmpvhd, err1 := disk.NewVirtualHardDisk(inst.WmiInstance)
+		if err1 != nil {
+			err = err1
+			return
+		}
+
+		vhdclone, err1 := tmpvhd.Clone()
+		if err1 != nil {
+			err = err1
+			return
+		}
+
+		retVhd, err1 := disk.NewVirtualHardDisk(vhdclone)
+		if err1 != nil {
+			err = err1
+			return
+		}
+
+		vhdpath, err1 := retVhd.GetPropertyHostResource()
+		if err1 != nil || len(vhdpath) == 0 {
```

Comment:
looks like you need if len(vhdpath) != 1 instead of 0.

Reply:
Done